conjur/logic/credential_provider/keystore_credentials_provider.py

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8 -*-
+
+"""
+KeyringCredentialsProvider module
+
+This module holds the logic for saving user-specific credentials
+to the system's keyring when they login
+"""
+
+# Builtins
+import logging
+
+# Internals
+from conjur.constants import KEYSTORE_ATTRIBUTES, MACHINE, LOGIN, PASSWORD
+from conjur.data_object import CredentialsData
+from conjur.errors import OperationNotCompletedException, CredentialRetrievalException
+from conjur.interface.credentials_store_interface import CredentialsStoreInterface
+from conjur.wrapper import KeystoreAdapter
+
+
+# pylint: disable=logging-fstring-interpolation
+class KeyringCredentialsProvider(CredentialsStoreInterface):
+    """
+    KeyringCredentialsProvider
+
+    This class holds logic for performing CRUD operations on credentials are kept
+    in the system's keystore
+    """
+
+    def __init__(self): # pragma: no cover
+        pass
+
+    # pylint: disable=line-too-long,logging-fstring-interpolation
```

Comment:
assign variable identifier = credential_data.machine and use it inside set_password